lib/rubocop/cop/rails/index_true.rb

```diff
@@ -0,0 +1,39 @@
+module RuboCop
+  module Cop
+    module Rails
+      # check that index: true is not being used in migrations
+      #
+      # @example
+      #   # bad
+      #   add_column :booking_templates, :booking_id, :integer
+      #
+      #   # good
+      #   add_index :booking_templates, :booking_id
+      #
+      class IndexTrue < Cop
+        MSG = '
+          index: true does not work in an add_column or change_column statement
+          please use add_index :table, :column
+        '.freeze
+
+        def_node_matcher :no_index_true, <<-PATTERN
+          (send nil {:add_column :change_column} $...)
```

Comment:
`(send nil {:add_column :change_column} _table _column _type (hash $...))`

Maybe you can refer to NotNullColumn cop's implementation.
https://github.com/bbatsov/rubocop/blob/26bad386fa8e3c85ad3a434faef0ec765ba612dc/lib/rubocop/cop/rails/not_null_column.rb